src/FSharpLint.Client/LSPFSharpLintServiceTypes.fsi

```diff
@@ -0,0 +1,50 @@
+module FSharpLint.Client.LSPFSharpLintServiceTypes
+
+type FSharpLintResponseCode =
+    | ToolNotFound = 1
+    | FileNotFound = 2
+    | FilePathIsNotAbsolute = 3
+    | CancellationWasRequested = 4
+    | DaemonCreationFailed = 5
+    | Version = 6
+
+type FSharpLintVersion = FSharpLintVersion of string
+
+type FSharpLintExecutableFile = FSharpLintExecutableFile of string
+
+type Folder = Folder of path: string
```

Comment:
@MrLuje how about we use the BCL's type for this, namely DirectoryInfo

Reply:
I like the fact that when you have a `Folder`, you know it's a valid/existing directory, which you can't be sure with DirectoryInfo (without calling `DirectoryInfo.Exists` before each usage)

Valid `Folder` wasn't really enforced but it can be done with something like this :
```fsharp
type Folder = private Folder of string
    with static member From (folder: string) =
            if System.IO.DirectoryInfo(folder).Exists then
                Folder folder |> Some
            else
                None
```

What do you think ?

Reply:
I like that! best of both worlds

Reply:
Changes pushed